### PR TITLE
Allow type declarations without capabilities

### PR DESCRIPTION
**Pull request by krishnangovindraj**
## Usage and product changes
Updates the grammar to allow standalone type-declarations.

## Implementation
This change arises from changing `mytype sub entity` to `entity mytype`. 
* Type annotations are now directly on the declaration  rather than on the `sub` capability.

Makes a few fields pub as well.

### CODE REVIEW

File: rust/parser/typeql.pest [hunk 1]
```diff
@@ -198,7 +198,7 @@ definable = { ( definition_type ~ SEMICOLON ) | definition_function | definition
 
 // TYPE DEFINITION =============================================================
 
-definition_type = { kind? ~ label ~ ( type_capability ~ ( COMMA ~ type_capability )* )? }
+definition_type = { kind? ~ label ~ (annotations | type_capability)? ~ ( COMMA ~ type_capability )* }
```
Comment (dmitrii-ubskii) [inline rust/parser/typeql.pest:201]:
```suggestion
definition_type = { kind? ~ label ~ ( ( annotations | type_capability ) ~ ( COMMA ~ type_capability )* )? }
```

**Review by dmitrii-ubskii — APPROVED** [review]

File: rust/parser/typeql.pest (hunk 1 above)
Comment (krishnangovindraj) [reply rust/parser/typeql.pest:201]: Will do. But my original formulation (and this one) has a drawback that there's no clean standard formatting anymore.
```
relation relates first-role,
    relates second-role;
````

Thoughts?

Comment (dmitrii-ubskii) [reply rust/parser/typeql.pest:201]: Cleared up in person, a small confusion about the grammar.

File: rust/parser/test/schema_queries.rs
Comment (krishnangovindraj) [inline rust/parser/test/schema_queries.rs:1]: @dmitrii-ubskii , Could you check these updates and see if any of the cases were intentional and need to be reverted?

File: rust/parser/test/mod.rs [hunk 2]
```diff
@@ -167,10 +167,10 @@ fn when_parsing_query_with_comments_they_are_ignored() {
     let query = r#"match
 # there's a comment here
 $x isa###WOW HERES ANOTHER###
-movie; count();"#;
+movie; count($x);"#;
```
Comment (dmitrii-ubskii) [inline rust/parser/test/mod.rs:170]: The equivalent of `count()` before is `count` without parentheses. This is not any more or less correct though.

File: rust/parser/test/schema_queries.rs [hunk 3]
```diff
@@ -49,11 +49,11 @@ evolves-final relates from-final as from;"#;
 #[test]
 fn test_define_query_with_plays_overrides() {
     let query = r#"define
-pokemon sub entity;
-evolves sub relation;
-evolves relates from,
+entity pokemon;
+relation evolves;
+relation evolves relates from,
     relates to;
```
Comment (dmitrii-ubskii) [inline rust/parser/test/schema_queries.rs:55]: `relation` is optional, but it's not incorrect to repeat.

File: rust/parser/test/schema_queries.rs
Comment (dmitrii-ubskii) [reply rust/parser/test/schema_queries.rs:1]: Looks correct top me, the changes stand.

**Pull request merged**